pkg/sql/colexec/external/external.go

```diff
@@ -647,3 +353,342 @@ func ScanFileData(param *ExternalParam, proc *process.Process) (*batch.Batch, er
 	bat.Cnt = 1
 	return bat, nil
 }
+
+func getData(bat *batch.Batch, Line []string, rowIdx int, param *ExternalParam) error {
```

Comment:
For this code, how much is shared with csv parsing? 

Can we think about if we can reduce this code using generics or something?

Reply:
I see, it is completely shared with csv.   Good.